test/resource_marketplace.test.js

```diff
@@ -182,6 +189,21 @@ describe('Marketplace Resource', function() {
       expect(errorThrown).to.be.true
       expect(errorMessage).to.equal('Error: Invalid offer: insufficient offer amount for listing')
     })
+
+    it('should throw an error if commission is not sufficient', async () => {
+      await marketplace.createListing(commissionListing)
+      await marketplace.makeOffer('999-001-1', invalidCommissionOffer)
+      let errorThrown = false
+      let errorMessage
+      try {
+        await marketplace.getOffer('999-001-1-0')
+      } catch(e) {
+        errorThrown = true
+        errorMessage = String(e)
+      }
+      expect(errorThrown).to.be.true
+      expect(errorMessage).to.equal('Error: Invalid offer: insufficient commission amount for listing')
```

Comment:
FYI, you can also use the chaiAsPromised package for these types of tests that an async function throws.
See https://github.com/OriginProtocol/origin-js/blob/master/test/service-data-store.test.js#L163

Reply:
Thanks. Yeah, I did look at that once. Can't remember if I had a good reason not to use it. Maybe a future refactor can use that.